sanic_jwt/authentication.py

```diff
@@ -194,45 +174,50 @@ def _get_token(self, request, refresh_token=False):
         return self._get_token(request, refresh_token=True)
 
     async def _get_user_id(self, user):
+        uid = self.config.user_id()
         if isinstance(user, dict):
-            user_id = user.get(self.config.user_id)
-        elif hasattr(user, 'to_dict'):
+            user_id = user.get(uid)
+        elif hasattr(user, "to_dict"):
             _to_dict = await utils.call(user.to_dict)
-            user_id = _to_dict.get(self.config.user_id)
+            user_id = _to_dict.get(uid)
         else:
             raise exceptions.InvalidRetrieveUserObject()
+
         return user_id
 
     async def get_access_token(self, user):
         payload = await self._get_payload(user)
         secret = self._get_secret(True)
         algorithm = self._get_algorithm()
 
-        return jwt.encode(payload, secret, algorithm=algorithm).decode('utf-8')
+        return jwt.encode(payload, secret, algorithm=algorithm).decode("utf-8")
 
     async def get_refresh_token(self, request, user):
-        refresh_token = utils.generate_token()
+        refresh_token = await utils.call(self.config.generate_refresh_token())
         user_id = await self._get_user_id(user)
         await utils.call(
             self.store_refresh_token,
             user_id=user_id,
             refresh_token=refresh_token,
-            request=request)
+            request=request,
+        )
         return refresh_token
 
     def is_authenticated(self, request, *args, **kwargs):
         try:
             is_valid, status, reasons = self.verify(request, *args, **kwargs)
         except Exception as e:
-            if self.config.debug:
+            if self.config.debug():
```

Comment:
We should probably document this. Originally I stuck this in for my convenience, but I think it has a legitimate use. And, since you started adding `logging we could push the exception to logs regardless of `DEBUG as a convenience.

Reply:
Looks like a nice idea. Perhaps we can use the message from the `Exception` for `logging.debug` OR even `raise exceptions.Unauthorized(msg)` if `isinstance(e, exceptions.SanicJWTException)` ?

Reply:
I was thinking to throw `Unauthorized` as is like now no matter what. Something vague. Then log the real exception.

Reply:
This ripped of one test case from tests, since I was trying to get to that `self.config.debug()` clause ... Will be available in the next push.